spark/src/main/java/org/apache/iceberg/spark/data/SparkOrcValueReaders.java

```diff
@@ -42,19 +42,26 @@
 import org.apache.spark.sql.types.Decimal;
 import org.apache.spark.unsafe.types.UTF8String;
 
-
-class SparkOrcValueReaders {
+public class SparkOrcValueReaders {
```

Comment:
I like the changes here and in `SparkOrcReader`, but they aren't really needed for this PR and increase the number of files that are touched. In general, I prefer keeping changes like this separate to avoid unnecessary conflicts, and so that we can revert commits without removing unrelated clean-up.

Reply:
I wanted to avoid duplicating the `if (precision < 18) then Decimal18Reader else Decimal38Reader` logic, since I am reusing it in `VectorizedSparkOrcReaders`. Similarly I made `utf8string` and `timestampTzs` methods public for reuse in the `vectorized` package. Let me know if you would like me to revert the Decimal change.

Reply:
Okay, I think that's a good justification for including this. Thanks!